test/e2e/gateway/005_request_header_modifier_test.go

```diff
@@ -18,241 +18,238 @@ package gateway
 import (
 	"net/http"
 
+	. "github.com/onsi/ginkgo"
 	"github.com/projectcontour/contour/test/e2e"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	gatewayv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
 )
 
-func testRequestHeaderModifierForwardTo(fx *e2e.Framework) {
-	t := fx.T()
-	namespace := "gateway-005-request-header-modifier-forward-to"
-
-	fx.CreateNamespace(namespace)
-	defer fx.DeleteNamespace(namespace)
-
-	fx.Fixtures.Echo.Deploy(namespace, "echo-header-filter")
-	fx.Fixtures.Echo.Deploy(namespace, "echo-header-nofilter")
-
-	route := &gatewayv1alpha1.HTTPRoute{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      "http-filter-1",
-			Labels:    map[string]string{"app": "filter"},
-		},
-		Spec: gatewayv1alpha1.HTTPRouteSpec{
-			Hostnames: []gatewayv1alpha1.Hostname{"requestheadermodifierforwardto.gateway.projectcontour.io"},
-			Gateways: &gatewayv1alpha1.RouteGateways{
-				Allow: gatewayAllowTypePtr(gatewayv1alpha1.GatewayAllowAll),
-			},
-			Rules: []gatewayv1alpha1.HTTPRouteRule{
-				{
-					Matches: []gatewayv1alpha1.HTTPRouteMatch{
-						{
-							Path: &gatewayv1alpha1.HTTPPathMatch{
-								Type:  pathMatchTypePtr(gatewayv1alpha1.PathMatchPrefix),
-								Value: stringPtr("/filter"),
-							},
-						},
+var _ = Describe("005-gateway-request-header-modifier", func() {
+	withHTTPGateway(func() {
+		Context("header modification on HTTPRoute rule ForwardTo", f.NamespacedTest(func(namespace string) {
+			It("works", func() {
```

Comment:
One of the things I don't like about Ginkgo is how much nesting it seems to lead to, just a whole lot of whitespace. WDYT about moving the bodies of the specs into their own functions that can be called from within the `It(...)`s? I think the downside is you have to explicitly pass any closure variables.

Reply:
yeah, the indenting gets especially bad with big go structs for the resources we create in k8s